common-content/en/module/js3/re-rendering/index.md

```diff
@@ -0,0 +1,95 @@
++++
+title = '🔁 Re-rendering'
+
+time = 10
+facilitation = false
+emoji= '🔁'
+[objectives]
+    1='Re-render a page based on user input'
+    2="Debug why a page isn't re-rendering as expected"
+[build]
+  render = 'never'
+  list = 'local'
+  publishResources = false
+
++++
+
+Now that we've shown we can log the search text, we can set the new value of the `searchTerm` state, and re-render the page.
+
+We should have a page like this:
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <title>Film View</title>
+  </head>
+  <body>
+    <input type="text" id="search" />
+    <template id="film-card">
+      <section>
+        <h3>Film title</h3>
+        <p data-director>Director</p>
+        <time>Duration</time>
+        <data data-certificate>Certificate</data>
+      </section>
+    </template>
+    <script>
+      const state = {
+        films: [
+          {
+            title: "Killing of Flower Moon",
+            director: "Martin Scorsese",
+            times: ["15:35"],
+            certificate: "15",
+            duration: 112,
+          },
+          {
+            title: "Typist Artist Pirate King",
+            director: "Carol Morley",
+            times: ["15:00", "20:00"],
+            certificate: "12A",
+            duration: 108,
+          },
+        ],
+        searchTerm: "",
+      };
+
+      const template = document.getElementById("film-card");
+      const createFilmCard = (film) => {
+        const card = template.content.cloneNode(true);
+        // Now we are querying our cloned fragment, not the entire page.
+        card.querySelector("h3").innerText = film.title;
+        card.querySelector("[data-director]").innerText = `Director: ${film.director}`;
+        card.querySelector("time").innerText = `${film.duration} minutes`;
+        card.querySelector("[data-certificate]").innerText = `Certificate: ${film.certificate}`;
```

Comment:
```suggestion
        card.querySelector("[data-certificate]").textContent = `Certificate: ${film.certificate}`;
```

Just because that's what it is on the video tbh